src/java.base/share/classes/sun/security/util/KeyUtil.java

```diff
@@ -403,5 +401,32 @@ public static byte[] trimZeroes(byte[] b) {
         return t;
     }
 
+    /**
+     * Finds the hash algorithm from an HSS/LMS public key.
+     *
+     * @param publicKey the HSS/LMS public key
+     * @return the hash algorithm
+     * @throws NoSuchAlgorithmException if key is from an unknown configuration
+     */
+    public static String hashAlgFromHSS(PublicKey publicKey)
```

Comment:
@ferakocz can you review this method? Thanks.

Reply:
Looks good to me.